### PR TITLE
Improve Radial Chart Functionality

### PR DESCRIPTION
**Pull request by mcnuttandrew**
This PR addresses a number of bugs that have been present in the radial chart for a while. Chief among them are: 

- When specify a variable such as opacity that happen to all have the same value, the value default to a small value. This problem is documented in #234 

- Radial chart not updating when new props are passed. This issue was briefly referenced in #161 but didn't have a direct gh-issue about it, so I added a example that demonstrates how to build a hover, as well demonstrates the bug. 

- Untangling the mouse events in radial (#157)

### CODE REVIEW

File: showcase/radial-chart/custom-radius-radial-chart.js [hunk 1]
```diff
@@ -23,17 +23,54 @@ import React from 'react';
 import RadialChart from 'radial-chart';
 
 export default class SimpleRadialChart extends React.Component {
+  constructor(props) {
```
Comment (Apercu) [inline showcase/radial-chart/custom-radius-radial-chart.js]: Prefer using the `state` class property.

File: showcase/radial-chart/custom-radius-radial-chart.js [hunk 2]
```diff
@@ -23,17 +23,54 @@ import React from 'react';
...
+    this.state = {
+      hoveredSection: false
+    };
+  }
+
   render() {
+    const {hoveredSection} = this.state;
     return (
       <RadialChart
+        animation={true}
```
Comment (Apercu) [inline showcase/radial-chart/custom-radius-radial-chart.js]: `true` is not necessary if the animation property is present

File: src/radial-chart/index.js [hunk 3]
```diff
@@ -116,7 +109,7 @@ class RadialChart extends React.Component {
    * @param {Object} event Event.
    * @private
    */
-  _triggerSectionHandler(handler, d, event) {
+  _sectionHandler(handler, d, event) {
```
Comment (Apercu) [inline src/radial-chart/index.js]: Same thing, don't need to check for the handler presence

File: src/radial-chart/index.js [hunk 4]
```diff
@@ -270,18 +237,28 @@ class RadialChart extends React.Component {
...
-                opacity: opacityFunctor && opacityFunctor(d),
-                stroke: strokeFunctor && strokeFunctor(d),
-                fill: fillFunctor && fillFunctor(d)
-              },
-              onMouseOver: e => this._sectionMouseOver(pieData[i], e),
-              onMouseOut: e => this._sectionMouseOut(pieData[i], e),
-              onClick: e => this._sectionClick(pieData[i], e),
-              key: i
-            }}/>)}
+            {data.map((d, i) => {
```
Comment (Apercu) [inline src/radial-chart/index.js]: I was enjoying the direct return of the jsx element in the map

File: src/radial-chart/index.js [hunk 5]
```diff
@@ -270,18 +237,28 @@ class RadialChart extends React.Component {
...
+              return (
+                <path {...{
+                  d: arc(pieData[i]),
+                  style: {
+                    opacity: opacityFunctor && opacityFunctor(d),
+                    stroke: strokeFunctor && strokeFunctor(d),
+                    fill: fillFunctor && fillFunctor(d)
+                  },
+                  onMouseEnter: e => {
```
Comment (Apercu) [inline src/radial-chart/index.js]: This could probably be kept on one line instead of 3

Comment (mcnuttandrew) [inline src/radial-chart/index.js]: true, but ill have to extend the max line width. truth be told i'd been waiting for an excuse to do so, so here we go

File: showcase/radial-chart/custom-radius-radial-chart.js (hunk 1 above)
Comment (mcnuttandrew) [inline showcase/radial-chart/custom-radius-radial-chart.js]: i'm not sure i know what you are referring to? can you point me to an example?

File: src/radial-chart/index.js (hunk 4 above)
Comment (mcnuttandrew) [inline src/radial-chart/index.js]: can you explain this comment a little more?

File: src/radial-chart/index.js (hunk 5 above)
Comment (Apercu) [inline src/radial-chart/index.js]: haha 👍

File: showcase/radial-chart/custom-radius-radial-chart.js (hunk 1 above)
Comment (Apercu) [inline showcase/radial-chart/custom-radius-radial-chart.js]:
```
class SimpleRadialChart extends Component {

  state = {
    hoveredSection: false
  }

}

File: src/radial-chart/index.js (hunk 4 above)
Comment (Apercu) [inline src/radial-chart/index.js]: `data.map((d, i)) => <path>`

File: .gitignore [hunk 6]
```diff
@@ -3,6 +3,7 @@ node_modules/
 .nyc_output/
 coverage/
 dist/
+showcase/node_modules/
```
Comment (Apercu) [inline .gitignore]: Isn't `node_modules/` already covering this one? That's louche!

Comment (mcnuttandrew) [inline .gitignore]: whoops didn't mean to leave this in, i was flailing around trying to make atom not search showcase/nodemodules

**Pull request merged**